functions/go/apply-replacements/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func main() {
+	if err := fn.AsMain(fn.ResourceListProcessorFunc(ApplyReplacements)); err != nil {
+		os.Exit(1)
+	}
+}
+
+func ApplyReplacements(rl *fn.ResourceList) error {
+	r := Replacements{}
+	return r.Process(rl)
+}
+
+type Replacements struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+const fnConfigKind = "ApplyReplacements"
+const fnConfigApiVersion = "fn.kpt.dev/v1alpha1"
+
+// Config initializes Replacements from a functionConfig fn.KubeObject
+func (r *Replacements) Config(functionConfig *fn.KubeObject) error {
+	if functionConfig.Kind() != fnConfigKind || functionConfig.APIVersion() != fnConfigApiVersion {
+		return fmt.Errorf("received functionConfig of kind %s and apiVersion %s, " +
+			"only functionConfig of kind %s and apiVersion %s is supported",
+			functionConfig.Kind(), functionConfig.APIVersion(), fnConfigKind, fnConfigApiVersion)
+	}
+	r.Replacements = []types.Replacement{}
+	if err := functionConfig.As(r); err != nil {
+		return fmt.Errorf("unable to convert functionConfig to %s:\n%w",
+			"replacements", err)
+	}
+	return nil
+}
+
+// Process implements framework.ResourceListProcessor interface.
+func (r *Replacements) Process(rl *fn.ResourceList) error {
+	if err := r.Config(rl.FunctionConfig); err != nil {
+		return err
+	}
+	transformedItems, err := r.Transform(rl.Items)
+	if err != nil {
+		return err
+	}
+	rl.Items = transformedItems
+	return nil
+}
+
+func (r *Replacements) Transform(items []*fn.KubeObject) ([]*fn.KubeObject, error) {
+	var transformedItems []*fn.KubeObject
+	var nodes []*yaml.RNode
+	for _, obj := range items {
```

Comment:
When https://github.com/GoogleContainerTools/kpt-functions-sdk/pull/526/files#r826210047 is done, we can use that instead of doing this conversion

Reply:
done.